### PR TITLE
Deincubate `ProviderFactory.gradleProperty`, `systemProperty`, `environmentVariable` and `fileContents`

### LINKED ISSUE #15575: Promote incubating `ProviderFactory.(gradleProperty|systemProperty|environmentVariable)`
### Summary

- org.gradle.api.provider.ProviderFactory#environmentVariable(java.lang.String)
- org.gradle.api.provider.ProviderFactory#environmentVariable(org.gradle.api.provider.Provider<java.lang.String>)
- org.gradle.api.provider.ProviderFactory#systemProperty(java.lang.String)
- org.gradle.api.provider.ProviderFactory#systemProperty(org.gradle.api.provider.Provider<java.lang.String>)
- org.gradle.api.provider.ProviderFactory#gradleProperty(java.lang.String)
- org.gradle.api.provider.ProviderFactory#gradleProperty(org.gradle.api.provider.Provider<java.lang.String>)

### Checklist
- [ ] Validate whether we should de-incubate the API in the 7.0 release.
    - If the removal is not possible, create a follow-up issue and link it in the [overview spreadsheet](https://docs.google.com/spreadsheets/d/19J1nR_dFKpfKdu5KDFMVZGfjR0ysT9DthsBUPwf8mkM/edit#gid=1195622786)
- [ ] Update release notes (add API to promoted features)
- [ ] Check User Manual (it might mention that the API is still incubating)
  - Think about updating snippets and samples to use this API
- [ ] Deprecate existing API that is replaced by the new one
  - If it's not immediately possible, create a follow-up issue with a milestone (e.g. `7.1 RC1` or `8.0 RC1`)
- [ ] Check Incubation Report on TeamCity ([example](https://builds.gradle.org/viewLog.html?buildId=40024670&buildTypeId=Gradle_Check_SanityCheck&tab=report_project951_Incubating_APIs_Report)) 
- [ ] Mark the story as done in the [overview spreadsheet](https://docs.google.com/spreadsheets/d/19J1nR_dFKpfKdu5KDFMVZGfjR0ysT9DthsBUPwf8mkM/edit?ts=5fcfefb8#gid=0).

### LINKED ISSUE #15576: Promote incubating `ProviderFactory.fileContents`
### Summary

- org.gradle.api.provider.ProviderFactory.fileContents(org.gradle.api.file.RegularFile)
- org.gradle.api.provider.ProviderFactory.fileContents(org.gradle.api.provider.Provider<org.gradle.api.file.RegularFile>)

### Checklist
- [x] Validate whether we should de-incubate the API in the 7.0 release.
    - If the removal is not possible, create a follow-up issue and link it in the [overview spreadsheet](https://docs.google.com/spreadsheets/d/19J1nR_dFKpfKdu5KDFMVZGfjR0ysT9DthsBUPwf8mkM/edit#gid=1195622786)
- [x] Update release notes (add API to promoted features)
- [ ] Check User Manual (it might mention that the API is still incubating)
  - Think about updating snippets and samples to use this API
- [ ] Deprecate existing API that is replaced by the new one
  - If it's not immediately possible, create a follow-up issue with a milestone (e.g. `7.1 RC1` or `8.0 RC1`)
- [ ] Check Incubation Report on TeamCity ([example](https://builds.gradle.org/viewLog.html?buildId=40024670&buildTypeId=Gradle_Check_SanityCheck&tab=report_project951_Incubating_APIs_Report)) 
- [ ] Mark the story as done in the [overview spreadsheet](https://docs.google.com/spreadsheets/d/19J1nR_dFKpfKdu5KDFMVZGfjR0ysT9DthsBUPwf8mkM/edit?ts=5fcfefb8#gid=0).

### PR DESCRIPTION
**Pull request by bamboo**
Fixes #15575
Fixes #15576

### CODE REVIEW

**Review by eskatos — COMMENTED** [review]
These should be added to the release notes

**Comment by bamboo** [comment]
> These should be added to the release notes

Yes, in fact that's the only reason the PR hasn't been merged yet :)

**Comment by bamboo** [comment]
@bot-gradle test this

File: subprojects/core-api/src/main/java/org/gradle/api/provider/ProviderFactory.java [hunk 1]
```diff
@@ -146,7 +140,6 @@
      *
      * @since 6.1
      */
-    @Incubating
     FileContents fileContents(RegularFile file);
```
Comment (eskatos) [inline subprojects/core-api/src/main/java/org/gradle/api/provider/ProviderFactory.java:143]: The `FileContents` interface should also be promoted

Comment (bamboo) [reply subprojects/core-api/src/main/java/org/gradle/api/provider/ProviderFactory.java:143]: Thanks for the review, @eskatos. `FileContents` was promoted in 3003017bd82bf91b56a2db42a6b07bc91bc0508b (and later added to the release notes).

**Pull request merged**